### DISCUSSION TITLE
Getting Trix editor and Alpine to work together

### DISCUSSION THREAD

**Discussion by taviks** [alpinejs/alpine#809] [4. General]
I'm struggling to get [Trix](https://trix-editor.org/) (the editor from Basecamp) and Alpine to work together. Has anyone done this? 

Here's an example: 
```html
<div x-data="{contentFromTextarea: null, contentFromTrix: null}">
  <!-- Trix through Alpina (not working) -->
  <div>
    <input id="trixTest" type="hidden" x-model="contentFromTrix">
    <trix-editor input="trixTest"></trix-editor>
    From trix:
    <span x-text="contentFromTrix"></span>
  </div>
</div>
```
Here's a jsfinddle for illustrating/testing: https://jsfiddle.net/8jk6pea5/2/

From looking in the Elements panel of Chrome's inspector, Trix is updating the hidden input as intended. So it seems the way it's getting updated isn't triggering the update in Alpine.

Note that moving `x-model="contentFromTrix"` to the `trix-editor` element _kinda_ works, but not really (some characters are deleted).

If there's no way to cleanly integrate Trix into Alpine I'll have to resort to using an event listener and manually updating things. e.g.

```js
document.addEventListener('trix-change', (event) => {
  // manually update Alpine data
})
```

I'd _really_ rather not go that route, however, so I'm hoping there's a clean approach.

Any insight?

**Comment by ryangjchandler** [comment]
I don't think Trix fires regular change or input events on the hidden input, so the simplest way would be using the `trix-change` event. That's how I've tackled it alongside Livewire before:

```html
<div x-data="{contentFromTextarea: null, contentFromTrix: null}">
  <!-- Trix through Alpina (not working) -->
  <div>
    <input id="trixTest" type="hidden" x-bind:value="contentFromTrix">
    <div x-on:trix-change="contentFromTrix = event.target.value">
        <trix-editor input="trixTest"></trix-editor>
    </div>
    From trix:
    <span x-text="contentFromTrix"></span>
  </div>
</div>
```

**Reply by taviks** [reply]
Thanks @ryangjchandler. That works!

Of course after beginning to implement this approach I find https://github.com/basecamp/trix/issues/680, which sadly seems like a dealbreaker. Now considering [Quill](https://quilljs.com/).

So anyone have any insight on which rich text editor to go with (especially if it plays particularly well with Alpine, though I suppose they all can), feel free to share!

**Reply by horizoncarlo** [reply]
Years later I wanted to say thanks @ryangjchandler as this helped me with Trix and Alpine - your approach works great!

**Comment by silicahd** [comment]
Here it is. The only problem is that this only work when you load the data in a modal. Unless anybody can figure out how init it at first. 

                  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/trix/1.3.1/trix.min.css" />


                   <div class="mb-2" x-data="{ description: @entangle('description').defer }"

                      x-init="$watch('description', function (value) {
                                 $refs.trix.editor.loadHTML(value)
                                 var length = $refs.trix.editor.getDocument().toString().length
                                 $refs.trix.editor.setSelectedRange(length - 1)
                                 }
                             )" wire:ignore>
         
                     <label class="form-label">  Rubrique Description <span class="text-danger">*</span></label>

                     @error('description')
                     <span class="error d-inline-block"><i class="mdi mdi-alert-circle"> </i> {{$message}}</span>
                     @enderror
                     <input name="description" id="description" type="hidden" x-model="description">
                     <div x-on:trix-change.debounce.1000ms="description = $refs.trix.value">
                         <trix-editor x-ref="trix" input="description" class="overflow-y-scroll"
                                      style="height: 20rem;"></trix-editor>
                     </div>
                 </div>
                
                
                
                <script src="https://cdnjs.cloudflare.com/ajax/libs/trix/1.3.1/trix.min.js"></script>
